src/ui/public/timefilter/timefilter.js

```diff
@@ -195,6 +195,20 @@ export const registerTimefilterWithGlobalState = _.once((globalState) => {
     timefilter.setTime(newTime);
     timefilter.setRefreshInterval(newRefreshInterval);
   });
+
+  const updateGlobalStateWithTime = () => {
+    globalState.time = timefilter.getTime();
+    globalState.refreshInterval = timefilter.getRefreshInterval();
+    globalState.save();
+  };
+
+  timefilter.on('refreshIntervalUpdate', () => {
```

Comment:
Since these listeners can be triggered outside a digest cycle, and they are calling the `globalState.save()` method, need to use `$rootScope.$listenAndDigestAsync()` to make sure that `globalState.save()` is not called outside a digest cycle.